### PR TITLE
Minor improvements to the use of StringMap/StringSet

### PR DESCRIPTION
**Pull request by jrose-apple**
- Remove handrolled StringSetIterator in favor of `keys()`
- Use llvm::StringSet instead of StringMap where appropriate
- Replace StringMap with DenseMap when the keys don't need to be owned

### CODE REVIEW

**Comment by jrose-apple** [comment]
Reviewers are mostly tagged to check that I didn't make a stupid mistake, particularly in components I don't spend much time in.

@swift-ci Please test

**Comment by jrose-apple** [comment]
@swift-ci Please test source compatibility

**Comment by jrose-apple** [comment]
@swift-ci Please test compiler performance

**Review by nkcsgexi — APPROVED** [review]
Good catch!

**Comment by aschwaighofer** [comment]
👍 for IRGen change

**Comment by jrose-apple** [comment]
@swift-ci please test

**Comment by jrose-apple** [comment]
@swift-ci Please test compiler performance

**Comment by jrose-apple** [comment]
@swift-ci Please test source compatibility

**Comment by jrose-apple** [comment]
@swift-ci Please test

**Comment by jrose-apple** [comment]
I can't win with this one. I guess I should have been more careful.

**Comment by jrose-apple** [comment]
@swift-ci Please test

**Comment by jrose-apple** [comment]
@swift-ci Please test source compatibility

File: lib/Serialization/SerializedModuleLoader.cpp [hunk 1]
```diff
@@ -376,19 +377,15 @@ FileUnit *SerializedModuleLoader::loadAST(
...
                  std::back_inserter(missing),
                  [&duplicates](const ModuleFile::Dependency &dependency)->bool {
       if (dependency.isLoaded() || dependency.isHeader())
         return false;
-      bool &seen = duplicates[dependency.RawPath];
-      if (seen)
-        return false;
-      seen = true;
-      return true;
+      return duplicates.insert(dependency.RawPath).second;
```
Comment (davidungar) [inline lib/Serialization/SerializedModuleLoader.cpp]: Clever!

**Pull request merged**